dirty_cat/similarity_encoder.py

```diff
@@ -228,7 +227,10 @@ def fit(self, X, y=None):
 
         n_samples, n_features = X.shape
         self.categories_ = list()
-        self.random_state_ = check_random_state(self.random_state)
+        if isinstance(self.random_state, np.random.RandomState):
+            self.random_state_ = copy.deepcopy(self.random_state)
```

Comment:
You should not do that: the semantics are that passing an initialized RandomState does not protect against side effects. Only passing a seed does.

Reply:
PS: the fact that your test is failing when you are passing an initialized RandomState shows that it is a good test :)